src/resource-archive/index.ts

```diff
@@ -178,7 +188,7 @@ class Watcher {
 
       // No need to capture the response of the top level page request
       const isFirstRequest = requestUrl.toString() === this.firstUrl.toString();
-      if (isLocalRequest && !isFirstRequest) {
+      if (isRequestFromAllowedDomain && !isFirstRequest) {
```

Comment:
Since we're now passing around the page URL, we could probably do away with the whole `isFirstRequest` logic altogether. But I'll save that for a future PR

Reply:
Hm yeah, perhaps!

Reply:
Honestly that was a hack to solve an early asset clobbering issue, but I think it's probably moot now. We could probably get rid of this later.